### PR TITLE
Add a noarch: python variant

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by tacaswell** [comment]
I'm not sure I understand the motivation here.  Presumably there is a run-time cost to not use cython so if that is available we want users to get the platform-specific version.

**Review by dopplershift — APPROVED** [review]

**Comment by dopplershift** [comment]
> I'm not sure I understand the motivation here. Presumably there is a run-time cost to not use cython so if that is available we want users to get the platform-specific version.

@tacaswell Right, that's exactly what's going on here. It's just adding an *additional* build and package that is noarch for Python versions (like free threading) that don't have support for the Cython portions of the package. All the platform-specific packages using Cython are still being built.

**Comment by tacaswell** [comment]
We trust that the solvers will always pick a platform specific version over noarch?

**Comment by isuruf** [comment]
> We trust that the solvers will always pick a platform specific version over noarch?

The platform specific version has `track_features` in it which makes the solver deprioritize that build because all conda solvers has a pass to minimize the number of features.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,18 +11,22 @@ source:
   sha256: 957f669d4922f92c171ba01bef7f29410668db09f6c02111e22b2bce446f3285
 
 build:
-  number: 0
-  skip: true  # [py2k]
+  number: 1
+  noarch: python   # [use_noarch]
+  track_features:
+    - fonttools_no_compile    # [use_noarch]
```
Comment (tacaswell) [inline recipe/meta.yaml:17]: @isuruf You mean the noarch version has the `treack_features`?  

Sorry if I am being difficult, I'm try to understand what is going on.

**Pull request merged**